shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -186,3 +188,13 @@ instance Crypto c => TranslateEra (AllegraEra c) WitnessSet where
 
 instance Crypto c => TranslateEra (AllegraEra c) Update where
   translateEra _ (Update pp en) = pure $ Update (coerce pp) en
+
+translateSnapShots ::
```

Comment:
This has been moved to EpochBoundary where SnapShots is defined. I have added some comments about what this is for. A copy of the comments I reproduce here.

-- | Translates (and completes) a SnapShots structure from one Era to another arbitrary Era.
--   Usefull in the translation phase from one Era (like ShelleyEra) to another (say AllegrEra).
--   A SnapShot is Era invariant, but the PulsingStakeDistr in the  _pstakeMark field of a
--   SnapShots, differs between Era as it contains an UTxO. By completing the  _pstakeMark field,
--   we remove the Era dependence, because the Completed  _pstakeMark field can live in any Era.
--   The consequence of this is that any Hardfork, any SnapsShots still pulsing, will be completed
--   at the hard fork. A similar strategy is used when serializing SnapShots.

Reply:
We used the same strategy to translate the pulsing reward update. Any pause because we are completing the pulsing will be lost in the cost of running the hard fork.